### PR TITLE
Run piped as a non-root user

### PR DESCRIPTION
**Pull request by nakabonne**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Part of https://github.com/pipe-cd/pipe/issues/987

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -133,7 +133,7 @@ container_pull(
     digest = "sha256:e357a240b1e8668c851a1dcb3e3244b4e037aab56029f394dda6925070d473a9",
     registry = "gcr.io",
     repository = "pipecd/piped-base",
-    tag = "0.1.0",
+    tag = "0.1.1",
```
Comment (nghialv) [inline WORKSPACE:136]: We are still using the old image.

File: dockers/piped-base/DOCKER_BUILD [hunk 2]
```diff
@@ -1,2 +1,2 @@
-version: 0.1.0
+version: 0.1.1
```
Comment (nghialv) [inline dockers/piped-base/DOCKER_BUILD:1]: I'm not sure why do we need to update this?

File: WORKSPACE (hunk 1 above)
Comment (nakabonne) [reply WORKSPACE:136]: Ah ok, now I can see this is the tag for base image.

File: dockers/piped-base/DOCKER_BUILD (hunk 2 above)
Comment (nakabonne) [reply dockers/piped-base/DOCKER_BUILD:1]: 👍

File: cmd/piped/BUILD.bazel [hunk 3]
```diff
@@ -18,9 +19,52 @@ go_binary(
     visibility = ["//visibility:public"],
 )
 
+# Create a passwd file with a root and pipecd user and uid.
```
Comment (nghialv) [inline cmd/piped/BUILD.bazel:22]: We have to load the needed functions before using them.

```
load(
    "@io_bazel_rules_docker//contrib:passwd.bzl",
    "passwd_entry",
    "passwd_file",
)

load(
    "@io_bazel_rules_docker//container:container.bzl",
    "container_image",
)
```

**Comment by nakabonne** [comment]
I've undone an unneeded tag update 👍

File: cmd/piped/BUILD.bazel [hunk 4]
```diff
@@ -18,9 +19,52 @@ go_binary(
...
+passwd_entry(
+    name = "root_user",
+    gid = 0,
+    uid = 0,
+    username = "root",
+    home = "/root",
+    info = "Root",
```
Comment (nghialv) [inline cmd/piped/BUILD.bazel:29]: nit: "root"

File: cmd/piped/BUILD.bazel [hunk 5]
```diff
@@ -18,9 +19,52 @@ go_binary(
...
+)
+
+passwd_entry(
+    name = "pipecd_user",
+    gid = 1000,
+    uid = 1000,
+    username = "pipecd",
+    home = "/home/pipecd"
+    info = "PipeCD User",
```
Comment (nghialv) [inline cmd/piped/BUILD.bazel:38]: nit: "pipecd"

File: cmd/piped/BUILD.bazel (hunk 3 above)
Comment (nakabonne) [reply cmd/piped/BUILD.bazel:22]: Damm... I still don't get the dependency structures in Bazel.

File: cmd/piped/BUILD.bazel (hunk 5 above)
Comment (nakabonne) [reply cmd/piped/BUILD.bazel:38]: Is the uppercase letter not preferable?

Comment (nghialv) [reply cmd/piped/BUILD.bazel:38]: Yes, I think we should use only the lowercase to reduce the unknown risks.

Comment (nakabonne) [reply cmd/piped/BUILD.bazel:38]: Gotcha!

**Comment by nghialv** [comment]
Looks good.
/approve

**Pull request merged**